src/units/key_manager.rs

```diff
@@ -71,17 +73,72 @@ impl KeyManager {
 
     async fn run(
         self,
-        cmd_rx: mpsc::UnboundedReceiver<ApplicationCommand>,
+        mut cmd_rx: mpsc::UnboundedReceiver<ApplicationCommand>,
     ) -> Result<(), crate::comms::Terminated> {
         let mut interval = tokio::time::interval(Duration::from_secs(5));
         interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
 
         // NOTE: We are not expecting any application commands right now.
-        let _ = cmd_rx;
-
         loop {
-            interval.tick().await;
-            self.tick().await;
+            let interval_tick = interval.tick();
+            let self_tick = self.tick();
+            pin_mut!(interval_tick);
+            pin_mut!(self_tick);
+
+            let ticks = select(interval_tick, self_tick);
+            pin_mut!(ticks);
+
+            let cmd = cmd_rx.recv();
+            pin_mut!(cmd);
+
+            match select(cmd, ticks).await {
+                Either::Left((cmd, _)) => match cmd {
+                    Some(ApplicationCommand::Terminate) | None => {
+                        return Err(Terminated);
+                    }
+                    Some(ApplicationCommand::RegisterZone {
+                        register: crate::api::ZoneAdd { name, .. },
+                    }) => {
+                        let cfg_path = self.dnst_keyset_data_dir.join(format!("{name}.cfg"));
+
+                        let status = std::process::Command::new(&self.dnst_keyset_bin_path)
+                            .arg("keyset")
+                            .arg("-c")
+                            .arg(&cfg_path)
+                            .arg("create")
+                            .arg("-n")
+                            .arg(name.to_string())
+                            .arg("-s")
+                            .arg(self.dnst_keyset_data_dir.join(format!("{name}.state")))
+                            .status();
+
+                        if let Err(_) = status {
+                            error!("[ZL]: Error creating keyset");
+                            continue;
+                        }
+
+                        // TODO: This should not happen immediately after
+                        // `keyset create` but only once the zone is enabled.
+                        // We currently do not have a good mechanism for that
+                        // so we init the key immediately.
+                        let status = std::process::Command::new(&self.dnst_keyset_bin_path)
```

Comment:
In future as we invoke `dnst` more times we will want to factor out calling it into a helper fn, and  that will also help with replacing it with a library later. We can hold off on adding that for now I guess.